en/docs/chapter_hashing/hash_map.md

```diff
@@ -513,25 +513,26 @@ The following code implements a simple hash table. Here, we encapsulate `key` an
 
 ## Hash collision and resizing
 
-Fundamentally, the role of the hash function is to map the entire input space of all keys to the output space of all array indices. However, the input space is often much larger than the output space. Therefore, **theoretically, there must be situations where "multiple inputs correspond to the same output"**.
+Essentially, the role of the hash function is to map the entire input space of all keys to the output space of all array indices. However, the input space is often much larger than the output space. Therefore, **theoretically, there will always be cases where "multiple inputs correspond to the same output"**.
 
-For the hash function in the above example, if the last two digits of the input `key` are the same, the output of the hash function will also be the same. For example, when querying for students with student numbers 12836 and 20336, we find:
+In the example above, with the given hash function, when the last two digits of the input `key` are the same, the hash function produces the same output. For instance, when querying two students with student IDs 12836 and 20336, we find:
 
 ```shell
 12836 % 100 = 36
 20336 % 100 = 36
 ```
 
-As shown in the figure below, both student numbers point to the same name, which is obviously incorrect. This situation where multiple inputs correspond to the same output is known as <u>hash collision</u>.
+As shown in the figure below, both student IDs point to the same name, which is obviously incorrect. This situation where multiple inputs correspond to the same output is called as <u>hash collision</u>.
 
 ![Example of hash collision](hash_map.assets/hash_collision.png)
 
-It is easy to understand that the larger the capacity $n$ of the hash table, the lower the probability of multiple keys being allocated to the same bucket, and the fewer the collisions. Therefore, **expanding the capacity of the hash table can reduce hash collisions**.
 
```

Comment:
There is still one more redundant line.

I would suggest checking the formatting in the files page https://github.com/krahets/hello-algo/pull/1358/files

Reply:
Resolved. Thanks for your reminding.